src/main/scala/tscfg/model.scala

```diff
@@ -16,61 +19,107 @@ object model {
 
       def %(t: Type): AnnType = AnnType(t, comments = Some(s))
     }
+
   }
 
   object durations {
+
     sealed abstract class DurationQualification
 
-    case object ns      extends DurationQualification
-    case object us      extends DurationQualification
-    case object ms      extends DurationQualification
-    case object second  extends DurationQualification
-    case object minute  extends DurationQualification
-    case object hour    extends DurationQualification
-    case object day     extends DurationQualification
+    case object ns extends DurationQualification
+
+    case object us extends DurationQualification
+
+    case object ms extends DurationQualification
+
+    case object second extends DurationQualification
+
+    case object minute extends DurationQualification
+
+    case object hour extends DurationQualification
+
+    case object day extends DurationQualification
+
   }
 
   sealed abstract class Type
 
   sealed abstract class BasicType extends Type
 
-  case object STRING    extends BasicType
-  case object INTEGER   extends BasicType
-  case object LONG      extends BasicType
-  case object DOUBLE    extends BasicType
-  case object BOOLEAN   extends BasicType
-  case object SIZE      extends BasicType
+  case object STRING extends BasicType
+
+  case object INTEGER extends BasicType
+
+  case object LONG extends BasicType
+
+  case object DOUBLE extends BasicType
+
+  case object BOOLEAN extends BasicType
+
+  case object SIZE extends BasicType
+
   case class DURATION(q: DurationQualification) extends BasicType
 
   val recognizedAtomic: Map[String, BasicType] = Map(
-    "string"    -> STRING,
-    "int"       -> INTEGER,
-    "integer"   -> INTEGER,
-    "long"      -> LONG,
-    "double"    -> DOUBLE,
-    "boolean"   -> BOOLEAN,
-    "size"      -> SIZE,
-    "duration"  -> DURATION(ms)
+    "string" -> STRING,
+    "int" -> INTEGER,
+    "integer" -> INTEGER,
+    "long" -> LONG,
+    "double" -> DOUBLE,
+    "boolean" -> BOOLEAN,
+    "size" -> SIZE,
+    "duration" -> DURATION(ms)
   )
 
   case class ListType(t: Type) extends Type
 
-  case class AnnType(t: Type,
-                     optional:      Boolean = false,
-                     default:       Option[String] = None,
-                     comments:      Option[String] = None
+  final case object AnnType{
+
+    val extendsPattern: Regex = """@\w+ extends ([a-zA-Z0-9]+)""".r
+
+    def isDefine(commentString: String): Boolean = commentString.trim.startsWith("@define")
+
+    def isParent(commentString: String): Boolean = commentString.trim.equals("@define_abstract")
```

Comment:
Please split `@define_abstract` so this is to be indicated with two separate tokens:`@define abstract`.  Reasons:
- you are already dealing with `@define extends` where the tokens as separated
- in a future version we may be adding, not only the already suggested `@define enum`, but also probably a variation of this "abstract" feature with for example `@define trait`.  (In fact, I was wondering if you would like to incorporate such variation here... but maybe too much for a single PR ;).

Reply:
I'm not 100% sure if I got your comment on this correctly but I tried to solve  this. Let me short explaining why I decided for the finally implemented solution.

My first approach result has been: 
```scala
isDefine(commentString) && commentString.trim.endsWith("abstract")
```

but this would allow for e.g. `@define_foo_bar_abstract` or `@define bar foo abstract`. To avoid this and enforce the syntax we would like to have I ended up with the finally implemented solution which I think might be more precise on what we are expecting here 

```scala 
  def isParent(commentString: String): Boolean = if (isDefine(commentString)) {
      commentString.replace(DEFINE_STRING, "").trim.equals("abstract")
    } else
      false
```

I hope I got your comment right here. If not, pls let me know.

Reply:
For your second suggestion on `@define trait`: I had the same idea during working on this PR. The only reason I did not add trait support (for now) was my uncertainty about the benefit of having the abstract class and trait support. 

I had the following concerns:
- trait support would be scala only
- I did not saw any benefit of the trait implementation over the abstract class implementation 

I'm open to add trait support to this PR as well if you think this would increase the benefit of the abstract object definition. Would you mind telling me what you think about these two concerns of me? Maybe I'm missing something?